src/amo/components/ReportAbuseButton/index.js

```diff
@@ -20,35 +21,43 @@ import { normalizeFileNameId, sanitizeHTML } from 'amo/utils';
 import Button from 'amo/components/Button';
 import DismissibleTextForm from 'amo/components/DismissibleTextForm';
 import type { OnSubmitParams } from 'amo/components/DismissibleTextForm';
-import type { AppState } from 'amo/store';
 import type { AddonAbuseState } from 'amo/reducers/abuse';
-import type { DispatchFunc } from 'amo/types/redux';
+import type { AppState } from 'amo/store';
 import type { AddonType } from 'amo/types/addons';
+import type { ElementEvent } from 'amo/types/dom';
 import type { I18nType } from 'amo/types/i18n';
-import { hasAbuseReportPanelEnabled } from 'amo/addonManager';
+import type { DispatchFunc } from 'amo/types/redux';
 
 import './styles.scss';
 
 type Props = {|
   addon: AddonType,
 |};
 
+type PropsFromState = {|
+  abuseReport: AddonAbuseState | Object,
```

Comment:
Because `mapStateToProps` can return `{}` for `abuseReport`. I tried to avoid making changes to code in this patch, and just keeping it to cleaning up types. Looking at this, we could change it to return `null` instead of `{}`, which I suppose is cleaner. This has been this way since 2017. I guess it makes sense to clean it up in this patch.